torchao/float8/config.py

```diff
@@ -48,12 +48,16 @@ def short_str(self):
 @dataclass(frozen=True)
 class CastConfig:
     """
-    Configuration for casting a single tensor to float8
+    Configuration for maybe casting a single tensor to float8
     """
 
     scaling_type: ScalingType = ScalingType.DYNAMIC
     scaling_granularity: ScalingGranularity = ScalingGranularity.TENSORWISE
     static_scale: Optional[torch.Tensor] = None
+    # If True, this tensor is not scaled to float8 and left in its original
+    # precision.
+    # TODO(ideally before this PR lands): a better name for this
+    keep_in_original_precision: bool = False
```

Comment:
IMO if we are configuring scaling via N different knobs (type, granularity, scale, etc) and we have "disabled" for one of the knobs, we also need "disabled" for all the other knobs to avoid inconsistencies.  Having the "disable" at the level of CastConfig seems like a cleaner way to configure this, although I don't love the name.

How about this:

```
Float8LinearConfig(...):
    # if cast config is specified, use it to scale/cast
    # if cast config is None, leave the tensor in original precision
    cast_config_input: Optional[CastConfig]
    ...
```

Reply:
```
Float8LinearConfig(...): # if cast config is specified, use it to scale/cast # if cast config is None, leave the tensor in original precision cast_config_input: Optional[CastConfig] ...
```

Realized that this conflicts with the current behavior of using `None` for the per-gemm-input override of input|weight|grad_output to mean "use the original config".  Going to take your suggestion instead, the inconsistency is minor.